### PR TITLE
[MI-743] migrate to app framework v2

### PR DESCRIPTION
**Pull request by joseconsador**
/cc @zendesk/mintegrations

### Description
Update the app to use apps framework v2.
Remove the order ID field ID custom field, as this will no longer be used.
Update app version number to 2.0.0 .
 
screen shot 2016-07-21 at 3 57 20 pm

The app still has the current design and functionality, except that it now uses `access_token` instead of an API key.

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-743
* App Scaffold: https://github.com/zendesk/app_scaffold/

### Risks
* [HIGH] App may not work with current users.

### CODE REVIEW

**Comment by miogalang** [comment]
:+1:

**Comment by samgavinio** [comment]
👍

**Comment by jmramos02** [comment]
👍

File: dist/assets/iframe.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <title>App Scaffold</title>
```
Comment (danielbreves) [inline dist/assets/iframe.html]: You may want to update this?

File: src/javascripts/legacy_app.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
+import BaseApp from 'base_app';
+
+var sprintf = require('sprintf-js').sprintf;
```
Comment (danielbreves) [inline src/javascripts/legacy_app.js]: We added `helpers.fmt` recently: https://github.com/zendesk/app_scaffold/blob/master/lib/javascripts/helpers.js

**Comment by danielbreves** [comment]
This is awesome! How did it go? Any feedback for us? 😺

File: package.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  "private": true,
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/zendesk/app_scaffold.git"
+  },
+  "scripts": {
+    "build": "webpack -p",
+    "watch": "webpack --watch"
+  },
+  "license": "Apache-2.0",
```
Comment (sandlerr) [inline package.json]: @miogalang better not do that

**Comment by sandlerr** [comment]
💥 awesome! This is a great candidate for not using the zendesk_proxy if I understand it right. No need to use `client.ajax` - you can just use `window.fetch()`, `$.ajax()` or even `new XMLHttpRequest()` or whatever ajax library you like.

**Comment by joseconsador** [comment]
@danielbreves @sandlerr 
Thanks for the review. We'll make sure to address those in the coming PRs. 

Aside from the issue with the [translations](https://github.com/zendesk/app_scaffold/pull/14), it was a good transition, the how to migrate doc was on point. It's great that we're no longer restricted to what libraries we might be using.

One thing I had to get used to was working around the asynchronous nature of the client API, which I think was different from v1. 

Overall, awesome stuff!

**Pull request merged**